### DISCUSSION TITLE
Non-TC Hindcasts?

### DISCUSSION THREAD

**Discussion by notstarboard** [StormSurgeLive/asgs#1074] [Q&A] [answered]
Hi Folks! Question here about hindcasting in ASGS. There's a big section on the wiki for TC hindcasting, and I know from our past discussions (and from Dave doing it!) that it's possible to manually set up hindcasts for past TCs. Is there any sort of a similar mechanism for hindcasting with historical GFS data?

It looks like the URLs from downloadGFS.sh and get_gfs.pl only have data back 10 days or so, so I'm not expecting it to be as simple as setting your cold start date to whenever you'd like. Is there a workaround for this if we're able to get our hands on the data? And if there were a stable, public url with more historical data available, do you think there would be any value in pulling from there if the user were to select a cold start date farther in the past?

**Comment by jasonfleming** [comment, marked as answer]
Hey @notstarboard the basic idea would be the same: set the coldstart date so that the tidal initialization ends at the beginning of your analysis period. Then place the archived data for the time range of interest on your filesystem for ASGS to find. It checks the local filesystem  for the data it needs before it attempts to download data from a remote source. NOAA does archive model data in various locations, with the time range of the archived data depending on the meteorological data source. You might also choose something other than GFS, such as FNL or ERA5.

**Reply by notstarboard** [reply]
That's really good to know! I didn't realize it would check for downloaded data before attempting to download its own. Makes sense!

**Comment by notstarboard** [comment]
This was more of a hypothetical at the time of my post, but now I'm actually trying to accomplish this. The goal is to try running with COAMPS forcing, since that provided very good results just recently with Hurricane Idalia. 

Am I correct in assuming that the local winds would need to be in grib2 format like the winds that you would typically download? It would be great if we could insert OWI ASCII winds, since that's what ASGS converts the grib2 files into anyway, but the code doesn't seem set up for that.

Also, I know that for TC mode hindcasts, you can step through to the next advisory by pointing the .dat symbolic link to the next best track file, and then updating the index-at.xml file to point to the next advisory. Is there a similar mechanism for non-TC-mode hindcasts? I can imagine a fort.222 symbolic link, or something, but if everything needs to be in grib2 format like I suspect I'm not sure how this would work.

Let me know your thoughts, and happy to have a discussion about support first! I'll be very busy until 9/18 due to comprehensive exams, but I'd be happy to chat after that. Thanks!

**Comment by jasonfleming** [comment]
Yes I believe we can handle this with the existing ASGS code as-is, or with some modest changes. I will send you and @wwlwpd an email.
